ruby/lib/fluent_tools/utils.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require 'rbconfig'
+
+module FluentTools
+  # Shared utilities for fluent-tools
+  module Utils
+    # Project constants
+    REPO_OWNER = 'Automattic'
+    BINARY_NAME = 'fluent-tools'
+    REPO_NAME = 'fluent-rust-tools'
+
+    # Supported platforms for cross-compilation
+    SUPPORTED_PLATFORMS = %w[
+      x86_64-linux
+      arm64-linux
+      x86_64-darwin
+      arm64-darwin
+      x86_64-windows
+      arm64-windows
+    ].freeze
+
+    # Platform to Rust target mapping
+    PLATFORM_RUST_TARGETS = {
+      'x86_64-linux' => 'x86_64-unknown-linux-gnu',
+      'arm64-linux' => 'aarch64-unknown-linux-gnu',
+      'x86_64-darwin' => 'x86_64-apple-darwin',
+      'arm64-darwin' => 'aarch64-apple-darwin',
+      'x86_64-windows' => 'x86_64-pc-windows-gnu',
+      'arm64-windows' => 'aarch64-pc-windows-gnullvm'
+    }.freeze
+
+    # Platform detection using RbConfig to identify the current system
+    # Returns a string in the format "{architecture}-{os}" (e.g., "arm64-darwin")
+    # Returns nil if the platform is not supported
+    def self.detect_platform
+      host_os = RbConfig::CONFIG['host_os']
+      host_cpu = RbConfig::CONFIG['host_cpu']
+
+      platform = case host_os
+                 when /linux/
+                   'linux'
+                 when /darwin/
+                   'darwin'
+                 when /mingw|mswin/
+                   'windows'
+                 else
+                   return nil
+                 end
+
+      architecture = case host_cpu
+                     when /x86_64|amd64/
+                       'x86_64'
+                     when /arm64|aarch64/
+                       'arm64'
+                     else
+                       return nil
+                     end
+
+      "#{architecture}-#{platform}"
+    end
+
+    # Generate the expected binary name for a given platform
+    # Adds .exe extension for Windows platforms
+    def self.binary_name_for_platform(platform)
+      if platform.include?('windows')
+        "#{BINARY_NAME}-#{platform}.exe"
+      else
+        "#{BINARY_NAME}-#{platform}"
+      end
```

Comment:
Let's use the helper you already have for that instead of duplicating their logic?
```suggestion
      extension = binary_extension(platform)
      "#{BINARY_NAME}-#{platform}#{extension}"
```